src/CfgMgr32/CfgMgr32+CM_NOTIFY_FILTER.cs

```diff
@@ -0,0 +1,182 @@
+            }
+
+            /// <summary>
+            /// Creates a <see cref="CM_NOTIFY_FILTER"/>, registering notifications for a device
+            /// interface.
+            /// </summary>
+            /// <param name="classGuid">
+            /// The <see cref="Guid"/> of the device interface class for which to receive notifications.
+            /// </param>
+            /// <returns>
+            /// A new instance of the <see cref="CM_NOTIFY_FILTER"/> struct.
+            /// </returns>
+            public static CM_NOTIFY_FILTER Create(Guid classGuid)
+            {
+                return new CM_NOTIFY_FILTER()
+                {
+                    cbSize = Marshal.SizeOf<CM_NOTIFY_FILTER>(),
+                    ClassGuid = classGuid,
+                    FilterType = CM_NOTIFY_FILTER_TYPE.CM_NOTIFY_FILTER_TYPE_DEVICEINTERFACE,
+                };
+            }
+
+            /// <summary>
+            /// Creates a <see cref="CM_NOTIFY_FILTER"/>, registering notifications for a device
+            /// instance.
+            /// </summary>
+            /// <param name="target">
+            /// A handle to the device for which to receive notifications.
+            /// </param>
+            /// <returns>
+            /// A new instance of the <see cref="CM_NOTIFY_FILTER"/> struct.
+            /// </returns>
+            public static CM_NOTIFY_FILTER Create(IntPtr target)
+            {
+                return new CM_NOTIFY_FILTER()
+                {
+                    cbSize = Marshal.SizeOf<CM_NOTIFY_FILTER>(),
+                    hTarget = target,
+                    FilterType = CM_NOTIFY_FILTER_TYPE.CM_NOTIFY_FILTER_TYPE_DEVICEHANDLE,
+                };
+            }
+
+            /// <summary>
+            /// Creates a <see cref="CM_NOTIFY_FILTER"/>, registering notifications for a device
+            /// instance.
+            /// </summary>
+            /// <param name="instanceId">
+            /// The device instance ID for the device for which to receive notifications.
+            /// </param>
+            /// <returns>
+            /// A new instance of the <see cref="CM_NOTIFY_FILTER"/> struct.
+            /// </returns>
+            public static CM_NOTIFY_FILTER Create(string instanceId)
+            {
+                if (instanceId == null)
+                {
+                    throw new ArgumentNullException(nameof(instanceId));
+                }
+
+                if (instanceId.Length > CfgMgr32.MAX_DEVICE_ID_LEN)
+                {
+                    throw new ArgumentOutOfRangeException(nameof(instanceId));
+                }
+
+                var filter = new CM_NOTIFY_FILTER()
+                {
+                    cbSize = Marshal.SizeOf<CM_NOTIFY_FILTER>(),
+                    FilterType = CM_NOTIFY_FILTER_TYPE.CM_NOTIFY_FILTER_TYPE_DEVICEINSTANCE,
+                };
+
+                Marshal.Copy(instanceId.ToCharArray(), 0, (IntPtr)filter.InstanceId, instanceId.Length);
```

Comment:
`ToCharArray()` copies the string to a newly allocated array, only for your `Marshal.Copy` to copy it again.
Can we copy it directly from the `string`'s own buffer instead?

Reply:
Could you give an example of how to access the `string`'s own buffer?

Reply:
I'll push a commit that shows it in this context.